crates/rust-analyzer/src/handlers/notification.rs

```diff
@@ -64,7 +64,14 @@ pub(crate) fn handle_did_open_text_document(
         if already_exists {
             tracing::error!("duplicate DidOpenTextDocument: {}", path);
         }
-        state.vfs.write().0.set_file_contents(path, Some(params.text_document.text.into_bytes()));
+        {
+            let vfs = &mut state.vfs.write().0;
+            vfs.set_file_contents(path.clone(), Some(params.text_document.text.into_bytes()));
+        }
```

Comment:
No need to do this (especially the path clone), I assume remnant from testing something

Reply:
I'll need to try removing these, but I'm *decently* confident that I needed to do this to drop the guard corresponding to `state.vfs.write()`.

It's probably going to be a moot point after I switch this over to the task-based approach.

Reply:
The temporary allocated for the write guard in `state.vfs.write().0.set_file_contents(path, Some(params.text_document.text.into_bytes()));` will be dropped immediately after the statement ends.